### DISCUSSION TITLE
Renovate creating MR only with schedule

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [renovatebot/renovate#15551] [Request Help]
Hello, I have the following configuration:

* GitLab - Renovate self-hosted

`config.js`
```
const npmHostRules = {
  hostType: "npm",
  matchHost: "******",
  username: ***,
  password: ***,
};

const dockerHostRules = {
  hostType: "docker",
  matchHost: "registry.gitlab.com",
  username: ***,
  password: ***,
};

const allDependenciesOneMergeRequest = {
  groupName: "all dependencies",
  separateMajorMinor: false,
  groupSlug: "renovate-dependencies",
  packageRules: [
    {
      matchPackagePatterns: ["*"],
      groupName: "all dependencies",
      groupSlug: "renovate-dependencies",
    },
  ],
  lockFileMaintenance: {
    enabled: true,
  },
};

const securityMergeRequestConfiguration = {
  vulnerabilityAlerts: {
    labels: ["security"],
  },
};

module.exports = {
  platform: "gitlab",
  gitAuthor: "GitLab Renovate Bot <****>",
  labels: ["renovate"],
  prConcurrentLimit: 1,
  automerge: false,
  enabledManagers: [],
  dependencyDashboard: true,
  hostRules: [npmHostRules, dockerHostRules],
  ...allDependenciesOneMergeRequest,
  ...securityMergeRequestConfiguration,
};
```

`renovate.json`

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "enabledManagers": [
    "npm",
    "nodenv",
    "nvm"
  ],
  "updateInternalDeps": true,
  "skipInstalls": false,
  "force": {
    "constraints": {
      "node": "16.14.0",
      "npm": "8.7.0"
    }
  },
  "packageRules": [
    {
      "matchPackageNames": [
        "node"
      ],
      "allowedVersions": "16.14.0"
    },
    {
      "matchPackageNames": [
        "npm"
      ],
      "allowedVersions": "8.7.0"
    }
  ]
}
```

And runs fine but somehow it created the dashboard with a schedule 

<img width="1010" alt="Renovate Issue Schedule" src="https://user-images.githubusercontent.com/36451129/167920933-a2c9f5ad-be9a-4d77-a28f-885d973f7bda.png">

However, the first Renovate execution on this project was without any `renovate.json` which caused Renovate to create an MR to add a `renovate.json` to the default branch, but after that, we added our custom `renovate.json` but still showed the schedule.

I ran it on debug mode and it shows the schedule which is not added to the `renovate.json`

```
DEBUG: Checking schedule(before 5am on monday, null) (repository****, branch=renovate/lock-file-maintenance)
DEBUG: Checking 1 schedule(s) (repository****, branch=renovate/lock-file-maintenance)
DEBUG: Checking schedule "before 5am on monday" (repository****, branch=renovate/lock-file-maintenance)
       "parsedSchedule": {"schedules": [{"t_b": [18000], "d": [2]}], "exceptions": [], "error": -1}
DEBUG: Package not scheduled (repository****, branch=renovate/lock-file-maintenance)
DEBUG: Skipping branch creation as not within schedule (repository****, branch=renovate/lock-file-maintenance)
DEBUG: Ensuring Dependency Dashboard (repository****)
```

**Questions**

1. Why is it showing a schedule? Is it because it ran initially without a configuration?
2. How can I make a renovate execution to fail is the project does not have a `renovate.json` in the default branch

**Comment by rarkins** [comment]
Lock file maintenance has a default schedule. You won't want it running non stop or it will create several PRs per day

**Reply by juancarlosjr97** [reply]
So, it has nothing to do with updates? The thing is the project has many updates available 

```
➜  *** git:(master) npm outdated
Package                Current    Wanted    Latest  Location
aws-sdk               2.1097.0  2.1133.0  2.1133.0  ***
eslint                  8.11.0    8.15.0    8.15.0  ***
eslint-plugin-import    2.25.4    2.26.0    2.26.0  ***
express                 4.17.3    4.18.1    4.18.1  ***
nodemon                 2.0.15    2.0.16    2.0.16  ***
```

And those are not getting created.

**Reply by rarkins** [reply]
Check the docs for rangeStrategy=update-lockfile

**Reply by juancarlosjr97** [reply]
Great, it makes sense. Thank you very much @rarkins !

**Reply by juancarlosjr97** [reply]
Regarding the possibility to make a project that is running renovate to fail executing renovate if does not have the `renovate.json` in the default branch is it possible?

**Reply by juancarlosjr97** [reply]
And is it possible to force to always create the PR and not wait to be scheduled as was happening in the project?

**Reply by rarkins** [reply]
I don't think I understand either of the last two questions. Can you add more details and also start a new discussion if they're separate topics?

**Reply by juancarlosjr97** [reply]
Sure @rarkins 

The second question is related to this same topic. When Renovate runs, it did not create a PR because "Lock file maintenance has a default schedule.", can I force renovate to create the MR always.

The reason is because we run renovate on schedule, not always, so it will not trigger the MR either way, right?

**Reply by juancarlosjr97** [reply]
The first question I started a separated topic https://github.com/renovatebot/renovate/discussions/15552 :)

**Reply by rarkins** [reply]
> Sure @rarkins
> 
> The second question is related to this same topic. When Renovate runs, it did not create a PR because "Lock file maintenance has a default schedule.", can I force renovate to create the MR always.

You will regret it if you run Renovate often, but you can disable the lock file maintenance schedule this way:

```
{
  "lockFileMaintenance": {
    "schedule": null
  }
}
```

**Reply by juancarlosjr97** [reply]
Great, exactly what I needed @rarkins!

Just to confirm, you are saying I am going to regret it because it is noisy, right?

**Reply by rarkins** [reply]
Yes, noise
